test/packetimpact/tests/syn_with_same_ip_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syn_with_same_ip_test
+
+import (
+	"golang.org/x/sys/unix"
+	"github.com/mohae/deepcopy"
+	"gvisor.dev/gvisor/pkg/tcpip/header"
+	tb "gvisor.dev/gvisor/test/packetimpact/testbench"
+	"testing"
+	"time"
+)
+
+const tcpLayerIndex int = 2
+const ipv4LayerIndex int = 1
+
+func TestSynWithSameLocalRemoteIP(t *testing.T) {
+	dut := tb.NewDUT(t)
+	listenFd, remotePort := dut.CreateListener(unix.SOCK_STREAM, unix.IPPROTO_TCP, 1)
+	defer dut.Close(listenFd)
+
+	conn := tb.NewTCPIPv4(t, dut, tb.TCP{DstPort: &remotePort}, tb.TCP{SrcPort: &remotePort})
+	defer conn.Close()
+
+	// Extracting dstaddr and configuring this as srcaddr
+	dstaddr := *conn.Outgoing[1].(*tb.IPv4).DstAddr
+	layersToSend := deepcopy.Copy(conn.Outgoing).(tb.Layers)
+	if err := layersToSend[ipv4LayerIndex].(*tb.IPv4).Mergelayer(tb.IPv4{SrcAddr: &dstaddr}); err != nil {
```

Comment:
You could just set the SrcAddr directly instead of using mergelayer.  I think like this:

`layersToSend[ipv4LayerIndex].(*tb.IPv4).SrcAddr = tb.IpAddress(dstaddr)`

Reply:
[GES] Ok will change accordingly.